### PR TITLE
Add SiStripCalCosmics ALCANANO

### PR DESCRIPTION
**Pull request by pieterdavid**
#### PR description:

This PR adds NanoAOD-style output for the SiStripCalCosmics ALCARECO (introduced in https://github.com/cms-sw/cmssw/pull/26032), based on the existing configuration (which was run by the DPG) to make flat trees for the Lorentz angle and backplane correction measurement, and taking advantage of the changes in https://github.com/cms-sw/cmssw/pull/33565 .
The code adds a plugin for per-run per-module information about the magnetic field etc., uses a `SimpleFlatTableProducer` for per-track information, and adds a plugin that stores the necessary per-cluster information (split in a base class that gets the clusters, and a subclass that stores the variables for the Lorentz angle and backplane correction - another subclass will be used for the gains tree).
This is a very small ALCARECO, so the size is small in both cases (the flat format is about the same size as the EDM format), but it has been extensively validated, and provides an interesting case to discuss the configuration and possibility of central production.
As a minimal solution to add this to cmsDriver I added a `SiStripCalCosmicsNano` ALCARECO based on the existing `SiStripCalCosmics`, and changed the code to insert a `NanoAODOutputModule` if the dataTier of a FilteredStream is NANOAOD. That seems to work (it can be tested with the `ALCA:SiStripCalCosmicsNano` step, which is also added to the relevant matrix workflows by this PR), but something less ad-hoc may be preferred - please provide input, this is the reason why the PR is marked as "work in progress".
The idea (discussed with XPOG) is to have this integrated for testing at T0 with CRAFT in October, so depending on if that works out and which release is used there, a backport to the 12_0 branch may be needed.

#### PR validation:

Event-by-event comparison for 2018 cosmics by @robervalwalsh , see [these slides](https://indico.cern.ch/event/992968/contributions/4424055/attachments/2270726/3856338/20210624_NanoAOD_CalibTrees.pdf)

CC: @robervalwalsh @mdelcourt @ataliercio @jandrejk @mmusich @tsusa @mariadalfonso @gouskos

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflows = 7.2, 7.21, 7.22, 7.23, 7.24, 7.3, 7.4, 134.813, 136.733, 138.1 and 138.2

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@pieterdavid this PR has now conflicts to solve.

**Comment by tvami** [comment]
+alca

File: CalibTracker/SiStripCommon/test/MakeCalibrationTrees/produceCalibrationTree_CosmicsLABP_cfg.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+process.GlobalTag = GlobalTag(process.GlobalTag, options.conditionGT, options.conditionOverwrite)
+
+process.load('FWCore.MessageService.MessageLogger_cfi')
+process.load('Configuration.StandardSequences.Services_cff')
+
+process.maxEvents = cms.untracked.PSet(input=cms.untracked.int32(options.maxEvents))
+
+process.source = cms.Source("PoolSource", fileNames=cms.untracked.vstring(options.inputFiles))
+if options.runNumber != -1:
+   if 'Cosmics' not in options.inputCollection:  # FIXME: this should be improved
```
[the remark this answers is not in the document]
Comment (tvami) [reply CalibTracker/SiStripCommon/test/MakeCalibrationTrees/produceCalibrationTree_CosmicsLABP_cfg.py:78]: @pieterdavid

[the remark this answers is not in the document]
Comment (pieterdavid) [reply CalibTracker/SiStripCommon/test/MakeCalibrationTrees/produceCalibrationTree_CosmicsLABP_cfg.py]: Sorry for the delay, yes, it's fine like this (I added the comment when I didn't understand these lines and thought more options were needed); it's a config for testing, so if you prefer I can also remove it

**Comment by pieterdavid** [comment]
Sorry for taking some time to update; I rebased on yesterday's IB and addressed a few comments raised by email. There is also a discussion on this planned in this afternoon's XPOG meeting https://indico.cern.ch/event/1066775/#3-sistrip-alcanano

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
+alca

**Comment by mariadalfonso** [comment]
+xpog

changes in PhysicsTools/NanoAOD/test/inspectNanoFile.py
see result here (the lumi block is now monitored also in the standard nano)
https://cms-nanoaod-integration.web.cern.ch/integration/34985/data106Xul17v2_size_report.html

**Comment by srimanob** [comment]
+Upgrade

**Comment by mmusich** [comment]
@pieterdavid looks like a new rebase is needed.

**Comment by qliphy** [comment]
@pieterdavid  Please resolve the conflict.

**Comment by pieterdavid** [comment]
Thanks @mmusich and @qliphy , rebased (the conflict was a minor one with https://github.com/cms-sw/cmssw/pull/35202 in the relval steps definitions)

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+alca

 * this was just rebased since the last signiture

**Comment by tvami** [comment]
Hi @cms-sw/pdmv-l2 colleagues, what's your take of this PR? It's been around for a while now :)

**Comment by kskovpen** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by tvami** [comment]
> +xpog
> 
> changes in PhysicsTools/NanoAOD/test/inspectNanoFile.py
> see result here (the lumi block is now monitored also in the standard nano)
> https://cms-nanoaod-integration.web.cern.ch/integration/34985/data106Xul17v2_size_report.html

@mariadalfonso would you mind signing this again? The PR has been rebased since you did it already

**Comment by mariadalfonso** [comment]
+xpog

**Comment by qliphy** [comment]
+1

**Comment by mmusich** [comment]
@perrotta @qliphy @tvami 
would it make sense to backport this to 12_0_X? 
CRAFT would be a perfect test bench.

**Comment by tvami** [comment]
I agree that it would be nice to have this for CRAFT.

**Pull request merged**